pkg/cloud/aws/aws.go

```diff
@@ -50,30 +41,19 @@ var (
 	ocpMetadataFile string
 )
 
-// AddAWSFlags adds basic flags needed by AWS.
-func AddAWSFlags(command *cobra.Command) {
-	command.Flags().StringVar(&infraID, infraIDFlag, "", "AWS infra ID")
-	command.Flags().StringVar(&region, regionFlag, "", "AWS region")
-	command.Flags().StringVar(&ocpMetadataFile, "ocp-metadata", "",
-		"OCP metadata.json file (or directory containing it) to read AWS infra ID and region from (Takes precedence over the flags)")
-	command.Flags().StringVar(&profile, "profile", aws.DefaultProfile(), "AWS profile to use for credentials")
-	command.Flags().StringVar(&credentialsFile, "credentials", aws.DefaultCredentialsFile(), "AWS credentials configuration file")
-}
-
 // RunOnAWS runs the given function on AWS, supplying it with a cloud instance connected to AWS and a reporter that writes to CLI.
 // The functions makes sure that infraID and region are specified, and extracts the credentials from a secret in order to connect to AWS.
-func RunOnAWS(restConfigProducer restconfig.Producer, gwInstanceType string,
-	function func(cloud api.Cloud, gwDeployer api.GatewayDeployer, reporter api.Reporter) error) error {
+func RunOnAWS(config *rest.Config, gwInstanceType string,
+	function func(cloud api.Cloud, gwDeployer api.GatewayDeployer, reporter api.Reporter) error, reporter api.Reporter) error {
 	if ocpMetadataFile != "" {
 		err := initializeFlagsFromOCPMetadata(ocpMetadataFile)
-		exit.OnErrorWithMessage(err, "Failed to read AWS information from OCP metadata file")
-	} else {
-		utils.ExpectFlag(infraIDFlag, infraID)
-		utils.ExpectFlag(regionFlag, region)
+		if err != nil {
+			return errors.Wrap(err, "Failed to read AWS information from OCP metadata file")
+		}
+	} else if infraID == "" || region == "" {
+		return errors.New("You must specify the infra-ID and/or region flag")
```

Comment:
These are actually unexported global vars so there's no way for the user to specify them 😄